docs/readme-espressif.md

```diff
@@ -42,34 +41,311 @@ cd boot/espressif/hal/esp-idf
+2. Bootloader calculates the SHA-256 hash digest of the public key and writes the result to eFuse.
+3. Bootloader validates the application images and prepare the booting process (MCUboot phase).
+4. Bootloader burns eFuse to enable Secure Boot V2.
+5. Bootloader proceeds to load the Primary image.
+
+After that the Secure Boot feature is permanently enabled and on every next boot the ROM bootloader will verify the MCUboot bootloader image.
+The process of an usual boot:
+
+1. On startup, the ROM bootloader checks the Secure Boot enable bit in the eFuse. If it is enabled, the boot will proceed as following.
+2. ROM bootloader verifies the bootloader's signature block integrity (magic number and CRC). Interrupt boot if it fails.
+3. ROM bootloader verifies the bootloader image, interrupt boot if any step fails.: \
+3.1. Compare the SHA-256 hash digest of the public key embedded in the bootloader’s signature block with the digest saved in the eFuses. \
+3.2. Generate the application image digest and match it with the image digest in the signature block. \
+3.3. Use the public key to verify the signature of the bootloader image, using RSA-PSS with the image digest calculated from previous step for comparison.
+4. ROM bootloader executes the bootloader image.
+5. Bootloader does the usual verification (MCUboot phase).
+6. Proceeds to boot the Primary image.
+
+## [Flash Encryption](#flash-encryption)
+
+The Espressif Flash Encryption is hardware-assisted, transparent to the MCUboot process and is an additional security measure beyond MCUboot existent features.
+The Flash Encryption implementation is also based on [IDF](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/security/flash-encryption.html) and is intended for encrypting off-chip flash memory contents, so it is protected against physical reading.
+
+When enabling the Flash Encryption, the user can encrypt the content either using a **device generated key** (remains unknown and unreadable) or a **host generated key** (owner is responsible for keeping the key private and safe). After the flash encryption gets enabled through eFuse burning on the device, all read and write operations are decrypted/encrypted in runtime.
+
+### [Building bootloader with Flash Encryption](#building-bootloader-with-flash-encryption)
+
+In order to build the bootloader with the feature on, the following configurations must be enabled:
+
+For **release mode**:
+```
+CONFIG_SECURE_FLASH_ENC_ENABLED=1
+CONFIG_SECURE_FLASH_ENCRYPTION_MODE_RELEASE=1
+```
+
+For **development mode**:
+```
+CONFIG_SECURE_FLASH_ENC_ENABLED=1
+CONFIG_SECURE_FLASH_ENCRYPTION_MODE_DEVELOPMENT=1
+```
+**ATTENTION**: on development phase is strongly recommended adding the following configuration in order to keep the debugging enabled and also to avoid any unrecoverable/permanent state change:
```

Comment:
```suggestion
:warning: ***ATTENTION***
On development phase is strongly recommended adding the following configuration in order to keep the debugging enabled and also to avoid any unrecoverable/permanent state change:
```
Keeping the same structure as the other warnings sections.